client/me/purchases/purchase-item/index.jsx

```diff
@@ -137,138 +159,145 @@ class PurchaseItem extends Component {
 		return null;
 	}
 
-	placeholder() {
-		return (
-			<span className="purchase-item__wrapper">
-				<div className="purchase-item__plan-icon" />
-				<div className="purchase-item__details">
-					<div className="purchase-item__title" />
-					<div className="purchase-item__purchase-type" />
-					<div className="purchase-item__purchase-date" />
-				</div>
-			</span>
-		);
-	}
+	getPurchaseType() {
+		const { purchase, site, translate, slug, showSite } = this.props;
+
+		if ( showSite ) {
+			return translate( '%(purchaseType)s for {{button}}%(site)s{{/button}}', {
+				args: {
+					purchaseType: purchaseType( purchase ),
+					site: site.name,
+				},
+				components: {
+					button: (
+						<button
+							className="purchase-item__site-name"
+							onClick={ ( event ) => {
+								event.stopPropagation();
+								event.preventDefault();
+								page( getPurchaseListUrlFor( slug ) );
+							} }
+							title={ translate( 'View subscriptions for %(siteName)s', {
+								args: {
+									siteName: site.name,
+								},
+							} ) }
+						/>
+					),
+				},
+			} );
+		}
 
-	scrollToTop() {
-		window.scrollTo( 0, 0 );
+		return purchaseType( purchase );
 	}
 
-	renderIcon() {
+	getPaymentMethod() {
 		const { purchase } = this.props;
 
-		if ( ! purchase ) {
-			return null;
-		}
+		if ( isRenewing( purchase ) ) {
+			const imgSrc =
+				purchase.payment.type === 'credit_card'
+					? getPaymentMethodImageURL( purchase.payment.creditCard.type )
+					: payPalImage;
```

Comment:
I'm not sure we should assume that any renewable payment method which isn't a credit card must necessarily be PayPal.  (It might be currently true -- to be honest I have no idea -- but even if it is, it's an assumption that could easily break in the future.)

I think for now maybe you could only display the PayPal info if `purchase.payment.type === 'paypal'` and just let anything else display as an empty payment method for the time being?

Reply:
That's a good call @DavidRothstein. Thanks for the suggestion.

Reply:
All cleaned up in this commit: https://github.com/Automattic/wp-calypso/pull/47610/commits/f6bce851add556769574688078b0489cbafaaafb